app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,36 @@
+<section>
+  <h1><%= @recipe.name %> </h1>
+  <% if @user_has_access %>
+    <section>
+      <p>Preparation time: <%= @recipe.preparation_time %> </p>
+      <p>cooking time: <%= @recipe.cooking_time %> </p>
+      <p>Steps:</p>
+      <p><%= @recipe.description %> </p>
+      <div>
+        <%= form_with(model: @recipe, url: recipe_path(@recipe), method: :patch, remote: true) do |f| %>
+          <% if @recipe.public %>
+            <%= value = 'Public'%>
+          <% else %>
+            <%= value = 'Private'%>
+          <% end %>
+          <%= f.hidden_field :public, value: !@recipe.public, disabled: true %>
+          <%if @recipe.public? %>
+            <%= f.submit 'Switch to private' %>
+          <% else %>
+            <%= f.submit 'Switch to public' %>
+          <% end %>
```

Comment:
- [ ] It would be nice if you add some authorization here because any person can make your recipe private who signs in this app :fearful: Kindly fix this issue to make your app professional :+1: